spyder/api/toolbar.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+spyder.api.toolbar
+==================
+
+Toolbar widgets designed specifically for Spyder plugin widgets.
+"""
+
+# Third party imports
+from qtpy.QtWidgets import (QApplication, QFrame, QHBoxLayout, QStyle,
+                            QWidget, QGridLayout, QVBoxLayout)
+
+# Local imports
+from spyder.utils.qthelpers import set_iconsize_recursively
+
+
+class SpyderPluginToolbar(QWidget):
+    """
+    Spyder plugin toolbar class.
+
+    All plugin widgets must use this class for their uppermost toolbar.
+    """
+    CLOSE_COL = 0
+    CONTENT_COL = 2
+    OPTIONS_COL = 4
+
+    def __init__(self, parent=None):
+        super(SpyderPluginToolbar, self).__init__(parent)
+        self.layout = QGridLayout(self)
+        self.layout.setHorizontalSpacing(0)
+
+        bottom_margin = QApplication.instance().style().pixelMetric(
+            QStyle.PM_ToolBarItemMargin)
+        self.layout.setContentsMargins(0, 0, 0, bottom_margin)
+
+    def _add_hboxlayout_at_row(self, row):
+        """Add a new QHBoxLayout at row row."""
+        row_layout = QHBoxLayout()
+        row_layout.setContentsMargins(0, 0, 0, 0)
+        row_layout.setSpacing(QApplication.instance().style().pixelMetric(
+            QStyle.PM_ToolBarItemSpacing))
+        colspan = 1 if row == 0 else 3
+        self.layout.addLayout(row_layout, row, self.CONTENT_COL, 1, colspan)
+
+    def _get_hboxlayout_at_row(self, row):
+        """
+        Return the QHBoxLayout at row row. Add one if it doesn't already
```

Comment:
row row?

Another way of writing these docstrings if you want to include the parameter is to write.


```python
"""
Return the QHBoxLayout at `row`. Add one if it doesn't already
"""
```